src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/BookingEntity.kt

```diff
@@ -152,6 +154,16 @@ interface BookingRepository : JpaRepository<BookingEntity, UUID> {
     nativeQuery = true,
   )
   fun findBookings(serviceName: String, status: BookingStatus?, probationRegionId: UUID?, pageable: Pageable?): Page<BookingSearchResult>
+
+  @Modifying
+  @Query("UPDATE BookingEntity b set b.status = :status where b.id = :bookingId")
+  fun updateBookingStatus(bookingId: UUID, status: BookingStatus)
+
+  @Query(
+    "SELECT * FROM bookings WHERE status IS NULL AND service='temporary-accommodation' ",
```

Comment:
Is there any way we can eager load the associations up front in such a way that Hibernate doesn't have to make all the subqueries? I think i've tried this in the past but couldn't make it do it, interested to maybe learn if this just isn't possible on a query-by-query basis.

Reply:
I hve tried it but getting 'org.hibernate.loader.MultipleBagFetchException: cannot simultaneously fetch multiple bags' . Looks like its not straight forward changes for this SQL.
I even dont want to change the entity to annotate @Eager Fetch.

Reply:
That `MultipleBagFetchException` error it the bane of my life!